libs/features/placeholders.js

```diff
@@ -1,20 +1,20 @@
+import { customFetch, getConfig } from '../utils/utils.js';
+
 const fetchedPlaceholders = {};
 
 const getPlaceholdersPath = (config, sheet) => {
   const path = `${config.locale.contentRoot}/placeholders.json`;
   const query = sheet !== 'default' && typeof sheet === 'string' && sheet.length ? `?sheet=${sheet}` : '';
   return `${path}${query}`;
 };
-
-const fetchPlaceholders = async (config, sheet) => {
-  const placeholdersPath = getPlaceholdersPath(config, sheet);
-  const { customFetch } = await import('../utils/helpers.js');
-
-  fetchedPlaceholders[placeholdersPath] = fetchedPlaceholders[placeholdersPath]
+const fetchPlaceholders = async ({ config, sheet, placeholderRequest, placeholderPath }) => {
+  const path = placeholderPath || getPlaceholdersPath(config, sheet);
+  fetchedPlaceholders[path] = fetchedPlaceholders[path]
     // eslint-disable-next-line no-async-promise-executor
     || new Promise(async (resolve) => {
-      const resp = await customFetch({ resource: placeholdersPath, withCacheRules: true })
-        .catch(() => ({}));
+      const resp = await placeholderRequest
+        || await customFetch({ resource: path, withCacheRules: true })
+          .catch(() => ({}));
```

Comment:
Having 5 indents is usually a sign that things might need to be simplified. I don't have a better suggestion right now, but this piece of logic seems to be more complicated than needed to me.

Reply:
I've moved this part around quite a bit, also trying to expose the placeholderRequest // `fetchedPlaceholders` on the window obj, that wasn't much clearer unfortunately.

However rather than passing the request around via params we can populate `window.mph` https://github.com/adobecom/milo/pull/2737

Which option would you prefer, keeping things as is - or moving `window.mph` to the utils & adding the placeholderRequest into `window.mph` ?

Reply:
Maybe something like:
```
const fetchPlaceholders = async ({ config, sheet, placeholderRequest, placeholderPath }) => {
  const path = placeholderPath || getPlaceholdersPath(config, sheet);

  fetchedPlaceholders[path] = fetchedPlaceholders[path] || new Promise(async (resolve) => {
    try {
      const resp = await (placeholderRequest || customFetch({ resource: path, withCacheRules: true }));
      const json = resp.ok ? await resp.json() : { data: [] };

      if (json.data.length === 0) { resolve({}); return; }

      const placeholders = json.data.reduce((plc, item) => {
        plc[item.key] = item.value;
        return plc;
      }, {});

      resolve(placeholders);
    } catch (error) {
      resolve({});
    }
  });

  return fetchedPlaceholders[path];
};
```
Maybe I was even a bit too cautious with the `try/catch` and it can be removed as well, reducing one more neccessary indent.

Reply:
Ah I see, done as far as the linting rules allow me to 👍